Project/Source/Components/ComponentParticleSystem.cpp

```diff
@@ -0,0 +1,305 @@
+	const char* emitterTypeComboCurrent = emitterTypeCombo[(int) emitterType];
+	ImGui::TextColored(App->editor->textColor, "Shape:");
+	if (ImGui::BeginCombo("##Shape", emitterTypeComboCurrent)) {
+		for (int n = 0; n < IM_ARRAYSIZE(emitterTypeCombo); ++n) {
+			bool isSelected = (emitterTypeComboCurrent == emitterTypeCombo[n]);
+			if (ImGui::Selectable(emitterTypeCombo[n], isSelected)) {
+				emitterType = (EmitterType) n;
+			}
+			if (isSelected) {
+				ImGui::SetItemDefaultFocus();
+			}
+		}
+		ImGui::EndCombo();
+	}
+	ImGui::Checkbox("Random Frame", &isRandomFrame);
+
+	ImGui::ResourceSlot<ResourceShader>("shader", &shaderID);
+
+	UID oldID = textureID;
+	ImGui::ResourceSlot<ResourceTexture>("texture", &textureID);
+
+	ResourceTexture* textureResource = App->resources->GetResource<ResourceTexture>(textureID);
+
+	if (textureResource != nullptr) {
+		int width;
+		int height;
+		glGetTextureLevelParameteriv(textureResource->glTexture, 0, GL_TEXTURE_WIDTH, &width);
+		glGetTextureLevelParameteriv(textureResource->glTexture, 0, GL_TEXTURE_HEIGHT, &height);
+
+		if (oldID != textureID) {
+			ComponentTransform2D* transform2D = GetOwner().GetComponent<ComponentTransform2D>();
+			if (transform2D != nullptr) {
+				transform2D->SetSize(float2(width, height));
+			}
+		}
+
+		ImGui::Text("");
+		ImGui::Separator();
+		ImGui::TextColored(App->editor->titleColor, "Texture Preview");
+		ImGui::TextWrapped("Size:");
+		ImGui::SameLine();
+		ImGui::TextWrapped("%d x %d", width, height);
+		ImGui::Image((void*) textureResource->glTexture, ImVec2(200, 200));
+		ImGui::InputInt("Xtiles: ", &Xtiles);
+		ImGui::InputInt("Ytiles: ", &Ytiles);
+		ImGui::InputFloat("Scale: ", &scale);
+		ImGui::InputFloat("Life: ", &particleLife);
+
+		if (ImGui::InputFloat("Speed: ", &velocity)) {
+			CreateParticles(maxParticles, velocity);
+		}
+
+		if (ImGui::InputFloat("MaxParticles: ", &maxParticles)) {
+			CreateParticles(maxParticles, velocity);
+		}
+
+		ImGui::ColorEdit4("InitColor##", initC.ptr());
+		ImGui::ColorEdit4("FinalColor##", finalC.ptr());
+	}
+}
+
+float3 ComponentParticleSystem::CreateVelocity() {
+	if (emitterType == EmitterType::CONE) {
+		ComponentTransform* transform = GetOwner().GetComponent<ComponentTransform>();
+		float x = (float(rand()) / float((RAND_MAX)) * 0.2) - 0.2;
+		float z = (float(rand()) / float((RAND_MAX)) * 0.2) - 0.2;
+		float y = (float(rand()) / float((RAND_MAX)) * 1.0);
+		return float3(x, y, z);
+	}
+	if (emitterType == EmitterType::SPHERE) {
+		ComponentTransform* transform = GetOwner().GetComponent<ComponentTransform>();
+		float x = (transform->GetGlobalPosition().x) + (float(rand()) / float((RAND_MAX)) * 2.0) - 1.0f;
+		float z = (transform->GetGlobalPosition().z) + (float(rand()) / float((RAND_MAX)) * 2.0) - 1.0f;
+		float y = (transform->GetGlobalPosition().y) + (float(rand()) / float((RAND_MAX)) * 2.0) - 1.0f;
+		return float3(x, y, z);
```

Comment:
Hardcoded values should be maked with a TODO so that we know what to change later.